pkg/agent/cluster.go

```diff
@@ -70,9 +71,34 @@ func NewCluster(ctx context.Context, assetDir, rendezvousIP, kubeconfigPath, ssh
 	czero := &Cluster{}
 	capi := &clientSet{}
 
-	authToken, err := FindAuthTokenFromAssetStore(assetDir)
-	if err != nil {
-		logrus.Fatal(err)
+	var authToken string
+	var err error
+
+	authToken, _ = FindAuthTokenFromAssetStore(assetDir) //nolint:errcheck
+
+	switch workflowType {
+	case workflow.AgentWorkflowTypeInstall:
+		// do nothing. The auth tokens do not expire.
+	case workflow.AgentWorkflowTypeAddNodes:
+		// authToken was not found in the asset store, retrieve it from cluster
+		if authToken == "" && kubeconfigPath != "" {
+			authToken, _ = gencrypto.GetAuthTokenFromCluster(ctx, kubeconfigPath)
+			if authToken == "" {
+				logrus.Fatal("Auth token secret not found in the cluster.")
+			}
+		}
+		// check if authToken is expired
+		expiryTime, err := gencrypto.ParseExpirationFromToken(authToken)
+		logrus.Debugf("Auth token is valid upto %s.", expiryTime) // token will be refreshed after Auth token is valid upto 
+		if err != nil {
+			return nil, err
+		}
+		if time.Now().After(expiryTime) {
+			logrus.Fatal("Auth token is expired. Re-run 'add-nodes' command to create new image files(ISO/PXE files).")
```

Comment:
This messge should be made more clear to suggest users that they need to regenerate a new ISO which will contain a new valid token, then boot that ISO and monitor the node again. Need suggestions to convey it in crisp and clear message.